CHANGES.md

```diff
@@ -1,6 +1,361 @@
+* adjusted the LLVM optimizer to assume math functions don't set C's `errno`
+
+Improvements to Compilation Times / Generated Code
+--------------------------------------------------
+* reduced `chpl` memory footprint for large programs via the compiler driver  
+  (see https://chapel-lang.org/docs/2.0/technotes/driver.html)
+
+Memory Improvements
+-------------------
+
+Tool Improvements
+-----------------
+* added `chpl-language-server`, a language server implementation for Chapel  
+  (see https://chapel-lang.org/docs/2.0/tools/chpl-language-server/chpl-language-server.html)
+* added support for user-defined lint rules in `chplcheck`  
+  (see https://chapel-lang.org/docs/2.0/tools/chplcheck/chplcheck.html)
+* improved how locations are reported by tools making use of the `chpl` parser
+* improved rendering of `chplcheck` lint rules in editors
+* improved rendering of return intents by `chpldoc` for some browsers
+
+Documentation Improvements
+--------------------------
+* merged 'ChapelIO' documentation into the 'IO' module's documentation  
+  (see https://chapel-lang.org/docs/2.0/modules/standard/IO.html)
+* fixed a browser-specific rendering issue with spacing of return intents
+* updated platform documentation for AWS to reflect current best practices  
+  (see https://chapel-lang.org/docs/2.0/platforms/aws.html)
+* improved InfiniBand documentation for clarity and to mention `pmi` launches
+  (see https://chapel-lang.org/docs/2.0/platforms/infiniband.html)
+* refreshed documentation concerning troubleshooting GASNet runs  
+  (see https://chapel-lang.org/docs/2.0/usingchapel/multilocale.html#troubleshooting)
+* clarified required conditions for split initialization  
+  (see https://chapel-lang.org/docs/2.0/language/spec/variables.html#split-initialization))
+* clarified required conditions for copy elision  
+  (see https://chapel-lang.org/docs/2.0/language/spec/variables.html#copy-elision)
+* improved test system documentation to cover more features
+  (see TODO)
+* improved documentation of enum constants to leverage new 'chpldoc' features
+* fixed typos in the `Math.sqrt()` documentation
+* fixed some cases where documentation and argument lists did not match
+* fixed a bug in the spec where certain statement forms lacked semicolons
+* fixed various typos and formatting issues
+
+Example Codes
+-------------
+
+Syntax Highlighting
+-------------------
+
+Configuration / Build / Packaging Changes
+-----------------------------------------
+* updated Chapel prereqs to require CMake 3.20, C++ 17, and those of LLVM 17  
+  (see https://chapel-lang.org/docs/2.0/usingchapel/prereqs.html)
+* refreshed the sample installation commands in the prerequisites docs  
+  (see https://chapel-lang.org/docs/2.0/usingchapel/prereqs.html#installation)
+* removed support for Python 3.7 from 'chpldoc'
+* disallowed building the compiler with AMD support when using the bundled LLVM
```

Comment:
IMO it's necessary to say "AMD GPU support" here since AMD makes CPUs as well and it's not about that part